src/dbnode/storage/namespace.go

```diff
@@ -598,24 +598,49 @@ func (n *dbNamespace) QueryIDs(
 	ctx context.Context,
 	query index.Query,
 	opts index.QueryOptions,
-) (index.QueryResults, error) {
+) (index.QueryResult, error) {
 	callStart := n.nowFn()
 	if n.reverseIndex == nil { // only happens if indexing is enabled.
 		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
-		return index.QueryResults{}, errNamespaceIndexingDisabled
+		return index.QueryResult{}, errNamespaceIndexingDisabled
 	}
 
 	if n.reverseIndex.BootstrapsDone() < 1 {
 		// Similar to reading shard data, return not bootstrapped
 		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
-		return index.QueryResults{}, xerrors.NewRetryableError(errIndexNotBootstrappedToRead)
+		return index.QueryResult{},
+			xerrors.NewRetryableError(errIndexNotBootstrappedToRead)
 	}
 
 	res, err := n.reverseIndex.Query(ctx, query, opts)
 	n.metrics.queryIDs.ReportSuccessOrError(err, n.nowFn().Sub(callStart))
 	return res, err
 }
 
+func (n *dbNamespace) AggregateQuery(
+	ctx context.Context,
+	query index.Query,
+	opts index.QueryOptions,
+	aggResultOpts index.AggregateResultsOptions,
+) (index.AggregateQueryResult, error) {
+	callStart := n.nowFn()
+	if n.reverseIndex == nil { // only happens if indexing is enabled.
+		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
+		return index.AggregateQueryResult{}, errNamespaceIndexingDisabled
+	}
+
+	if n.reverseIndex.BootstrapsDone() < 1 {
+		// Similar to reading shard data, return not bootstrapped
+		n.metrics.queryIDs.ReportError(n.nowFn().Sub(callStart))
+		return index.AggregateQueryResult{},
+			xerrors.NewRetryableError(errIndexNotBootstrappedToRead)
+	}
+
+	res, err := n.reverseIndex.AggregateQuery(ctx, query, opts, aggResultOpts)
+	n.metrics.queryIDs.ReportSuccessOrError(err, n.nowFn().Sub(callStart))
```

Comment:
nit: add a metric for `aggregateQuery` (similar to `queryIDs`)

Reply:
(and reuse in all calls in this function)

Reply:
Nice one :+1: